lib/chef/provider/windows_task.rb

```diff
@@ -156,11 +176,12 @@ def action_delete
       def action_end
         if current_resource.exists
           Chef::Log.debug "#{new_resource} task exists"
-          if current_resource.status != :running
+          if current_resource.task.status != "running"
```

Comment:
we may DRY up this status check thing. may include some methods like `running?` , `ready?`, etc and can the same elsewhere.